cpp/include/cudf/detail/gather.cuh

```diff
@@ -518,6 +506,58 @@ void gather_bitmask(table_view const& source,
   }
 }
 
+template <typename MapItRoot>
+struct column_gatherer_impl<struct_view, MapItRoot> {
+  std::unique_ptr<column> operator()(column_view const& column,
+                                     MapItRoot gather_map_begin,
+                                     MapItRoot gather_map_end,
+                                     bool nullify_out_of_bounds,
+                                     cudaStream_t stream,
+                                     rmm::mr::device_memory_resource* mr)
+  {
+    structs_column_view structs_column(column);
+    auto gather_map_size{std::distance(gather_map_begin, gather_map_end)};
+    if (gather_map_size == -1) {
```

Comment:
When would this distance ever be negative?

Reply:
Yikes, this does not look right. I think this is another case of an errant keystroke in Vim. (`Ctrl-X` decrements `0` to `-1`, instead of cutting text.)
I'll fix this and re-test, when adding tests.

Reply:
I've fixed this now.

Reply:
Fixed.